File: 315 hw1/Codes/duru_ant.php
<?php
// Defining the function to use later to print key,value pairs
function foo($arr, $key){
    if(array_key_exists($key, $arr)){
        echo $key, "    ", $arr[$key];
        echo "\n";
    }    

}



// Initializing the associative array
$myArr = array(
            "first" => 10,
            "second" => 65,
            "third" => 4,
            "fourth" => 0,
            "fifth" => 77);
echo "---------------1-ASSOCIATIVE ARRAY HAS BEEN INITIALIZED.----------------";



// Getting and printing a value by its key
echo "\n";
echo "---------------2-GETTING THE VALUE OF KEY \"third\".----------------";
$myValue = $myArr["third"];
echo "\n";
echo "key: third, value: " , $myValue;



// Adding a new element with key "sixth" and value 6
echo "\n";
echo "---------------3-ADDING A NEW ELEMENT.----------------";
$myArr["sixth"] = 6;
echo "\n";
echo "A new element with key \"sixth\" has been added, its value is: ", $myArr["sixth"];



// Removing an element
echo "\n";
echo "---------------4-REMOVING AN ELEMENT.----------------";
echo "\n";
unset($myArr["second"]);
echo "The element with key \"second\" has been removed.";



// Modifying the value of an element
echo "\n";
echo "---------------5-MODIFYING THE VALUE OF AN ELEMENT.----------------";
echo "\n";
echo "The current value of the key \"first\" is: ", $myArr["first"];
$myArr["first"] = 56;
echo "\n";
echo "After the modify, the value of the key \"first\" is: ", $myArr["first"];



// Search for existence of a key
echo "\n";
echo "---------------6-SEARCHING EXISTENCE OF KEY \"second\".----------------";
echo "\n";
if( array_key_exists("second", $myArr)){
    echo "the key \"second\" is in the associative array with value: ", $myArr["second"];
}
else{
    echo "the key \"second\" is not in the associative array";
}




// Search for existence of a value
echo "\n";
echo "---------------7-SEARCHING EXISTENCE OF VALUE 77.----------------";
echo "\n";
if( array_search(77, $myArr) != FALSE){
    echo "77 is in the associative array with key: ", array_search(77, $myArr);
}




// Loop through the array and printing the pairs using my function
echo "\n";
echo "---------------8-PRINTING PAIRS.----------------";
echo "\n";
foreach($myArr as $key => $value){
    foo($myArr, $key);
}
?>

File: 315 hw1/Codes/duru_ant.py

# Implementing my function to use it later to print the pairs
def foo(k, d):
    print(k, d[k])


# Initializing the dictionary
my_dict = { 
        "first": 10,
        "second": 65,
        "third": 4,
        "fourth": 0,
        "fifth": 77 }
print("--------1-DICTIONARY HAS BEEN INITIALIZED---------")

# Getting the value for a given key
print("--------2-GETTING THE VALUE WITH \"third\" KEY---------")
my_value = my_dict["third"]
print("key: third, value: ", my_value)

# Adding element to the dictionary with key "sixth" and value 6
print("--------3-ADDING AN ELEMENT---------")
my_dict["sixth"] = 6
print("value ", my_dict["sixth"], " is added to the dictionary with key \"sixth\"")

# Removing element from the dictionary
print("--------4-REMOVING THE ELEMENT WITH KEY \"fourth\"---------")
my_dict.pop("second")
print("element with key \"second\" has been removed")

# Modifying an element
print("--------5-MODIFYING THE ELEMENT WITH KEY \"third\"---------")
print("current value of element with key \"first\" is: ", my_dict["first"])
my_dict["first"] = 56
print("After modify value of element with key \"first\" is: ", my_dict["first"])

# Searching for the existence of a key
print("--------6-SEARCHING FOR THE EXISTENCE OF A KEY---------")
if (my_dict.get("second") != None):
    print("the key \"second\" is in the dictionary with value: ", my_dict["second"])
else:
    print("the key \"second\" is not in the dictionary")
    

# Searching for the existence of a value
print("--------7-SEARCHING FOR THE EXISTENCE OF A VALUE---------")
if(list(my_dict.keys())[list(my_dict.values()).index(77)] != None):
    print("The value 77 is in the dictionary with key: ", list(my_dict.keys())[list(my_dict.values()).index(77)])
else:
    print("The value 77 is not in the dictionary")
    
# Loop to print pairs
print("--------8-PRINTING PAIRS---------")
for k in my_dict:
    foo(k, my_dict)
